optimum/habana/transformers/generation/utils.py

```diff
@@ -2290,11 +2300,10 @@ def expand_if_needed(tensor, new_size, value, dim=-1):
                 if (
                     self.generation_config.early_stopping
                     and is_min_length_reached
-                    and cur_len % self.generation_config.early_stopping_interval == 0
                     and num_eos_tokens >= num_beams_tensor
                 ):
                     break
-                elif cur_len == self.generation_config.max_length:
+                elif stopping_criteria(input_ids[:, :token_idx], scores):
```

Comment:
This slicing here will break static_shapes, right?

Reply:
why? same logic is here https://github.com/huggingface/optimum-habana/blob/main/optimum/habana/transformers/generation/utils.py#L1423, the 2297 line makes the output length incorrect. the self.generation.max_length is not equal to generation.max_length outside beam_search.

Reply:
Can we use stopping_criteria(input_ids, scores) instead of using input_ids[:, :token_idx] ?
That way we will still maintain static shape across runs and not impacting the perf.